### PR TITLE
{Core} --add option in Generic Update Operation: add warning info and help info

### LINKED ISSUE #17610: No detailed description for JSON format for --add option
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xfz11**
resolve #17610  #17586
The --add option in generic update operation is very hard to use, so more information about the generic update arguments is added to the help text. 
If option value is JSON string and its format is wrong, the error message should help users to fix it. Two cases are in consideration:
1, The JSON string is not escaped or has the wrong format: show the JSON parse error
2, The JSON string is parsed as a JSON array: in most case, --add option should pass a JSON Object instead of a JSON array. Passing a list may cause DeserializationError

## Help text
Add links(https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively?view=azure-cli-latest#generic-update-arguments) to the documentation page of cli generic update arguments
**Before**
image

**After**
image

## Error message
Add hints of the cause of error
**Before**
image

**After**
image

image

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -374,6 +374,7 @@ def __call__(self, parser, namespace, value, option_string=None):
     cli_ctx.register_event(EVENT_INVOKER_PRE_LOAD_ARGUMENTS, add_subscription_parameter)
 
 
+ge_update_arg_ref = 'https://aka.ms/AAby4o3'
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/arm.py:377]: Why `AAby4o3`?

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 2]
```diff
@@ -542,12 +544,14 @@ def add_properties(instance, argument_values, force_string):
                 try:
                     argument = shell_safe_json_parse(argument)
                 except (ValueError, CLIError):
-                    pass
+                    json_parse_fail_args.append(argument)
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/arm.py:547]: `shell_safe_json_parse` already has the exception handling logic.

https://github.com/Azure/azure-cli/blob/f34ca47e441c949c75df07d3bfa8e76e4f5d7a28/src/azure-cli-core/azure/cli/core/util.py#L513-L531

File: src/azure-cli-core/azure/cli/core/commands/command_operation.py [hunk 3]
```diff
@@ -236,11 +239,17 @@ def handler(self, command_args):  # pylint: disable=too-many-locals, too-many-st
...
+        try:
+            if self.setter_arg_name == 'parameters':
+                result = cached_put(self.cmd, setter, **setterargs)
+            else:
+                result = cached_put(self.cmd, setter, setterargs[self.setter_arg_name],
+                                    setter_arg_name=self.setter_arg_name, **setterargs)
+        except SerializationError:
+            if json_parse_fail_args:
+                logger.warning("JSON parse fail for '--add' argument value '%s'. If you want to pass a JSON string, "
+                               "read more in %s", ' '.join(json_parse_fail_args), ge_update_arg_ref)
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/command_operation.py:251]: You may directly show the exception raised by `shell_safe_json_parse` as warning. No need to make your own warning message.

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 1 above)
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/arm.py:377]: `ge` may impact the readability of the code.

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 2 above)
Comment (xfz11) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: --add option allows users to pass a string, not necessarily a JSON string. So warning will not show until DeserializationError is thrown. As your advice, the error raised by `shell_safe_json_parse ` has been reserved for generating warning message in latest code version

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 1 above)
Comment (xfz11) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: changed to cligenericupdate

File: src/azure-cli-core/azure/cli/core/commands/command_operation.py [hunk 4]
```diff
@@ -166,8 +168,9 @@ def handler(self, command_args):  # pylint: disable=too-many-locals, too-many-st
         from knack.util import CLIError
         from azure.cli.core.commands import cached_get, cached_put, _is_poller
         from azure.cli.core.util import find_child_item, augment_no_wait_handler_args
-        from azure.cli.core.commands.arm import add_usage, remove_usage, set_usage,\
+        from azure.cli.core.commands.arm import add_usage, remove_usage, set_usage, \
```
Comment (xfz11) [inline src/azure-cli-core/azure/cli/core/commands/command_operation.py:171]:
```suggestion
        from azure.cli.core.commands.arm import add_usage, remove_usage, set_usage,\
```

**Comment by jsntcy** [comment]
Close it as it's out of date.

**Pull request closed without merging**